appinventor/components/src/com/google/appinventor/components/runtime/CloudDB.java

```diff
@@ -524,11 +557,89 @@ private void checkAccountNameProjectIDNotBlank(){
       throw new RuntimeException("CloudDB ProjectID property cannot be blank.");
     }
   }
+
+  private String getFileExtension(String fullName) {
+    String fileName = new File(fullName).getName();
+    int dotIndex = fileName.lastIndexOf(".");
+    return dotIndex == -1 ? "" : fileName.substring(dotIndex + 1);
+  }
   
   private Jedis getJedis(){
     Jedis jedis = new Jedis("128.52.179.76", 6379);
     jedis.auth("test6789");
     return jedis;
   }
+
+  private YailList readFile(String fileName) {
```

Comment:
Please comment Jeff's code with the appropriate citation of ownership/contribution.